### PR TITLE
#220 Zoom link generation functionality attached to meeting creation

### PR DESCRIPTION
**Pull request by YuraRiabov**
Screenshot_692
Screenshot_693
Screenshot_694
Screenshot_697

### CODE REVIEW

**Review by va-dym-ko-val — APPROVED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/ZoomService.cs [hunk 1]
```diff
@@ -37,8 +38,12 @@ public async Task<CredentialsDto> GetNewCredentials(NewCredentialsRequestDto new
         return await GetCredentials(queryString);
     }
 
-    public async Task CreateZoomMeeting(Meeting meeting)
+    public async Task CreateZoomMeeting(long meetingId)
     {
+        var meeting = _context.Meetings
+            .Include(m => m.Author)
+                .ThenInclude(u => u.Credentials)
+            .First(m => m.Id == meetingId);
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/ZoomService.cs]: FirstOrDefauls and check for null

File: frontend/src/app/modules/bookings/bookings-page/bookings-page.component.html [hunk 2]
```diff
@@ -12,7 +12,7 @@
                     </div>
 
                     <div class="buttonRow">
-                        <button class="buttonZoom">Open {{ item.location }} Link</button>
+                        <button class="buttonZoom" (click)="followMeetingLink(i)">Open {{ item.location }} Link</button>
```
Comment (tatianahutii) [inline frontend/src/app/modules/bookings/bookings-page/bookings-page.component.html]: I believe we can simplify this logic

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Comment by tatianahutii** [comment]
Add screenshots

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/MeetingService.cs [hunk 3]
```diff
@@ -94,6 +97,11 @@ public async Task CreateMeeting(SaveMeetingDto meetingDto)
             await _context.Meetings.AddAsync(meeting);
 
             await _context.SaveChangesAsync();
+
+            if (meeting.LocationType == LocationType.Zoom)
+            {
+                await _zoomService.CreateZoomMeeting(meeting.Id);
+            }
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/MeetingService.cs]: it would be better to create a separate method with switch/case for this

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

**Review by tatianahutii — APPROVED** [review]

**Review by apofis96 — APPROVED** [review]

**Pull request merged**